### DISCUSSION TITLE
Mimicking "dynamic" behaviour for particular layouts

### DISCUSSION THREAD

**Discussion by master-of-m** [i3/i3#6100] [Q&A]
![layout](https://github.com/i3/i3/assets/91836055/2a39403f-0d5a-4846-a331-4b707aafee67)

Hi everybody, I know i3 isn't a dynamic wm, but I was wondering if it is possibile to automate it at least for simple layouts like the one in the picture.
I know you can save layouts but 1) they are fixed 2) they work with applications classes.

The behavior I want to achieve is the following:
1) when I create a new window in a new workspace it takes all the desk space as normal;
2) when I open the second window it creates a column of a x:y ratio instead of the default 50:50
3) when I open the third window it splits the right column horizontally in half

Is it possible to do this without too much of a hustle (which could mean switching wm is better)?
If yes, how should I proceed?

Thank you in advance!

**Comment by okraits** [comment]
Maybe you can use one of these scripts as a base:

* https://github.com/nwg-piotr/autotiling
* https://github.com/chmln/i3-auto-layout

Alternatively you could write such a script yourself, using the i3 IPC interface, which might be easy or more complex, depending on how sophisticated your script should be.
